src/bdhost/Util.cpp

```diff
@@ -77,25 +81,33 @@ namespace bdhost
     return json["size"].asUInt();
   }
 
-  bool SetReservedSpace(const uint64_t size)
+  std::string SetReservedSpace(const uint64_t size)
   {
-    static uint64_t contractSize = bdcontract::ContractRepository::Load(bdhost::Options::contract.c_str())->Size()/(1024 * 1024);
+    static uint64_t contractSize = bdcontract::ContractRepository::Load(bdhost::Options::contract.c_str())->Size();
 
-		auto reservedSpace = GetReservedSpace();
+    auto reservedSpace = GetReservedSpace();
 
     if(contractSize < (size + reservedSpace))
     {
       printf("Reserve size exceeds contractSize\n");
-      return false;
+      return "";
     }
 
-		Json::Value json;
-		json["size"] = Json::Value::UInt(size + reservedSpace);
+    Json::Value json;
+    json["size"] = Json::Value::UInt(size + reservedSpace);
 
     std::ofstream out(RESERVED_FILE);
-		out << json.toStyledString();
-		out.close();
+    out << json.toStyledString();
+    out.close();
+
+    std::string reserve_id = uuidgen();
+    mkdir(reserve_id.c_str(), 0755);
```

Comment:
This should go to the storage folder. Current working directory may be /usr/bin.